### PR TITLE
improve log levels

### LINKED ISSUE #1276: review log levels for different logs
After #1272 gets merged we need to review our log level.
I would like very much to get traces from logs when tx fails to apply during catchup, but also I don't care about traces from connection errors in Overlay module. Both of these use ERROR log level.

We should either change the less important ones from ERROR to WARNING and change in-memory push level to ERROR

or

we could change the more imporant ones to FATAL and leave in-memory push level as it is.

I prefer first solution.

### PR DESCRIPTION
**Pull request by vogel**
Fixes #1276

Signed-off-by: Rafał Malinowski <rafal.przemyslaw.malinowski@gmail.com>

### CODE REVIEW

File: src/history/HistoryManagerImpl.cpp [hunk 1]
```diff
@@ -84,7 +84,7 @@ HistoryManager::initializeHistoryArchive(Application& app, std::string arch)
     }
     if (getHas->getState() == Work::WORK_SUCCESS)
     {
-        CLOG(ERROR, "History") << "History archive '" << arch
+        CLOG(FATAL, "History") << "History archive '" << arch
```
Comment (zbyszkol) [inline src/history/HistoryManagerImpl.cpp]: Is this really FATAL?

File: src/history/HistoryManagerImpl.cpp [hunk 2]
```diff
@@ -202,7 +202,7 @@ HistoryManager::checkSensibleConfig(Config const& cfg)
 
     if (badArchives)
     {
-        CLOG(ERROR, "History") << "History archives misconfigured.";
+        CLOG(FATAL, "History") << "History archives misconfigured.";
```
Comment (zbyszkol) [inline src/history/HistoryManagerImpl.cpp]: Is this really FATAL?

File: src/historywork/ApplyLedgerChainWork.cpp [hunk 3]
```diff
@@ -132,13 +132,15 @@ ApplyLedgerChainWork::applyHistoryOfSingleLedger()
...
-            throw std::runtime_error(
+            auto msg =
                 fmt::format("replay of {:s} failed to connect on hash of LCL "
                             "predecessor {:s}",
                             LedgerManager::ledgerAbbrev(hHeader),
                             LedgerManager::ledgerAbbrev(
                                 lclHeader.header.ledgerSeq - 1,
-                                lclHeader.header.previousLedgerHash)));
+                                lclHeader.header.previousLedgerHash));
+            CLOG(ERROR, "History") << msg;
```
Comment (zbyszkol) [inline src/historywork/ApplyLedgerChainWork.cpp]: Anti-pattern? Leave it to some handler, don't log twice. Pros: file and line number of the error. Maybe we can use some mechanism to contain line numbers in exceptions.

File: src/process/ProcessManagerImpl.cpp [hunk 4]
```diff
@@ -158,7 +158,7 @@ ProcessExitEvent::Impl::run()
     assert(!mProcManagerImpl->isShutdown());
     if (mRunning)
     {
-        CLOG(ERROR, "Process") << "ProcessExitEvent::Impl already running";
+        CLOG(FATAL, "Process") << "ProcessExitEvent::Impl already running";
```
Comment (zbyszkol) [inline src/process/ProcessManagerImpl.cpp]: FATAL?

**Comment by MonsieurNicolas** [comment]
PR #1272 was rolled back ; #1326 is now tracking this type of problem

**Pull request closed without merging**